lib/core/sdam/server.js

```diff
@@ -158,6 +158,10 @@ class Server extends EventEmitter {
     // setup listeners
     this.s.pool.on('parseError', parseErrorEventHandler(this));
 
+    this.s.pool.on('drain', err => {
```

Comment:
is this for everything, or just legacy?

Reply:
this event is only emitted by the pool when it is _not_ in `legacyCompatMode` (i.e. when the unified topology is using it). We only need to listen to the event with unified's `Server`